R/quo-unquote.R

```diff
@@ -1,4 +1,133 @@
-#' Process unquote operators in a captured expression.
+#' Quasiquotation of an expression
+#'
+#' @description
+#'
+#' Quasiquotation is the mechanism that makes it possible to program
+#' flexibly with [tidyeval][tidy-evaluation] grammars like dplyr. It
+#' is enabled in all tidyeval functions, the most fundamental of which
+#' are [quo()] and [expr()].
+#'
+#' Quasiquotation is the combination of quoting an expression while
+#' allowing immediate evaluation (unquoting) of part of that
+#' expression. We provide both syntactic operators and functional
+#' forms for unquoting.
+#'
+#' - `UQ()` and the `!!` operator unquote their argument. It gets
+#'   evaluated immediately in the surrounding context.
+#'
+#' - `UQE()` is like `UQ()` but retrieves the expression of
+#'   [quosureish][is_quosureish] objects. It is a shortcut for `!!
+#'   get_expr(x)`.
```

Comment:
Use this with care: it is potentially unsafe because it loses the environment of the quosure